### PR TITLE
Support haproxy_exporter for Prometheus

### PR DESCRIPTION
**Pull request by sdickenson**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -34,3 +34,11 @@ haproxy_enable_syslog: false
 haproxy_use_graylog_json_logging: false
 haproxy_graylog_server:
 haproxy_graylog_server_port:
+
+# Prometheus exporter support for haproxy v1.8
+haproxy_external_exporter_enabled: false
+haproxy_exporter_download_url: https://github.com/prometheus/haproxy_exporter/releases/download/v0.12.0/haproxy_exporter-0.12.0.linux-amd64.tar.gz
```
Comment (smiller171) [inline defaults/main.yml:40]: Would it be better to get the version from its own variable?

File: defaults/main.yml [hunk 2]
```diff
@@ -34,3 +34,11 @@ haproxy_enable_syslog: false
...
+haproxy_exporter_listen_port: 9101
+haproxy_exporter_stats_user: admin
+haproxy_exporter_stats_password: password
```
Comment (smiller171) [inline defaults/main.yml:43]: Do I even wanna know why there's a default password of `password` here?

File: handlers/main.yml [hunk 3]
```diff
@@ -8,3 +8,6 @@
 
 - name: restart rsyslog-haproxy
   service: name=rsyslog state=restarted
+
+- name: reload systemd
+  command: /bin/systemctl daemon-reload
```
Comment (smiller171) [inline handlers/main.yml]: Prefer built-in modules when available

```suggestion
- name: reload systemd
  systemd:
    daemon_reload: yes

```

**Review by smiller171 — CHANGES_REQUESTED** [review]

**Review by rsanchez-s — APPROVED** [review]
looks good.

File: defaults/main.yml (hunk 2 above)
Comment (sdickenson) [reply defaults/main.yml:43]: No

File: defaults/main.yml (hunk 1 above)
Comment (sdickenson) [reply defaults/main.yml:40]: I didn't care to make this too flexible. This isn't necessary in haproxy v2+ so I don't see this being around forever.

Comment (smiller171) [reply defaults/main.yml:40]: fair enough

**Review by smiller171 — APPROVED** [review]
lgtm

**Pull request merged**